include/cutlass/arch/memory.h

```diff
@@ -514,35 +562,27 @@ void shared_load(void *dst, uint32_t ptr);
 template <>
 CUTLASS_DEVICE
 void shared_load<2>(void *dst, uint32_t ptr) {
-#if !defined(CUTLASS_ENABLE_SYCL) || defined(__SYCL_CUDA_ARCH__)
+#if defined(__CUDA_ARCH__) || defined(__SYCL_CUDA_ARCH__)
   asm volatile("ld.shared.u16 %0, [%1];\n"
     : "=h"(*reinterpret_cast<uint16_t *>(dst))
     : "r"(ptr));
+#else
+  CUTLASS_INVALID_CONTROL_PATH(
+      "Attempting to use Nvidia-specific code path on non-Nvidia hardware.");
 #endif
 }
 
 /// ld.shared - 32b
 template <>
 CUTLASS_DEVICE
 void shared_load<4>(void *dst, uint32_t ptr) {
-#if !defined(CUTLASS_ENABLE_SYCL) || defined(__SYCL_CUDA_ARCH__)
+#if defined(__CUDA_ARCH__) || defined(__SYCL_CUDA_ARCH__)
   asm volatile("ld.shared.u32 %0, [%1];\n"
     : "=r"(*reinterpret_cast<uint32_t *>(dst))
     : "r"(ptr));
-#endif
-}
-
-/// ld.shared - 64b
-template <>
-CUTLASS_DEVICE
-void shared_load<8>(void *dst, uint32_t ptr) {
```

Comment:
Why is this code deleted?

Reply:
:sweat_smile: oops - i blame my lack of neovim skills